### DISCUSSION TITLE
When installing k3s, how to customize the timeout time of each component

### DISCUSSION THREAD

**Discussion by Heyz6** [k3s-io/k3s#6620] [Q&A]
I would like to be able to customize the 5s timeout; see the log below
example：
Dec 08 15:31:19 VM-10-44-centos k3s[31183]: E1208 15:31:19.594118   31183 leaderelection.go:325] error retrieving resource lock kube-system/kube-controller-manager: Get "https://127.0.0.1:6444/apis/coordination.k8s.io/v1/namespaces/kube-system/leases/kube-controller-manager?timeout=5s": net/http: request canceled (Client.Timeout exceeded while awaiting headers)

[k3s-log.txt](https://github.com/k3s-io/k3s/files/10183492/k3s-log.txt)

**Comment by brandond** [comment]
That is not something that is configurable. If that connection is failing, it indicates a problem with networking between the servers in your cluster, between the servers and the datastore, or (if you are also seeing datastore latency warnings) a problem with the performance of your disk.

These messages from the log you attached indicate that the problem is probably the latter - due to poor disk performance:
```
Dec 08 15:31:19 VM-10-44-centos k3s[31183]: I1208 15:31:19.419074   31183 trace.go:205] Trace[2014813712]: "GuaranteedUpdate etcd3" type:*coordination.Lease (08-Dec-2022 15:31:14.419) (total time: 4999ms):
Dec 08 15:31:19 VM-10-44-centos k3s[31183]: Trace[2014813712]: [4.9991865s] [4.9991865s] END
```

Writes to the datastore should not take more than several hundred ms to complete at most. Anything higher than that indicates performance problems.
